lib/parser.js

```diff
@@ -85,14 +109,31 @@
         }
       }
 
-      // Escapes; we're either in double quotes or in unqouted text; all escapes apply equally.
-      if(str[i] == '\\') {
+      // Characters; we're either in double quotes or in unqouted text; all escapes apply equally.
+      if(str[i] === '\\') {
         if(i >= str.length - 1) {
           return retPartial(new Error("Trailing '\\' without escape"), i);
         }
-        // consume the backslash
+        // consume the escape character
         i++;
-        // Here we go, backslash escaping
+        // As a special case, always allow quoteCharacters, rawQuoteCharacters,
+        // and spaceCharacters to be escaped
+        var doubleQuoteChar = findElement(self.doubleQuoteCharacters, str[i]);
```

Comment:
Does updating `doubleQuoteChar` here let you get away with something funny? If `doubleQuoteCharacters = ['"','$']`, would this let `"\$$` parse to `$` instead of a mismatched quote error?